src/main/java/com/uber/profiling/YamlConfigProvider.java

```diff
@@ -168,6 +168,15 @@ private static void addConfig(Map<String, Map<String, List<String>>> config, Str
             for (Object entry : (Object[])value) {
                 configValueList.add(entry.toString());
             }
+         } else if (value instanceof Map) {
+            for (Object mapKey : ((Map)value).keySet()) {
+                String propKey = mapKey.toString();
+                Object propValue = ((Map)value).get(propKey);
+                if (propValue != null) {
+                    addConfig(config, override, key + "." + propKey, propValue);
+                 }
+            }
+            configMap.values().removeIf(List::isEmpty);
```

Comment:
Would you remove this line "configMap.values().removeIf(List::isEmpty)"? We could keep empty list in configMap, so we could distinguish between empty list (has the config key but no value) and none (do not have the config key).